homeassistant/components/econet/__init__.py

```diff
@@ -1 +1,153 @@
-"""The econet component."""
+"""Support for EcoNet products."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyeconet import EcoNetApiInterface
+from pyeconet.equipment import EquipmentType
+from pyeconet.errors import InvalidCredentialsError, PyeconetError
+
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import API_CLIENT, DOMAIN, EQUIPMENT
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["binary_sensor", "sensor", "water_heater"]
+PUSH_UPDATE = "econet.push_update"
+
+INTERVAL = timedelta(minutes=60)
+
+
+async def async_setup(hass, config):
+    """Set up the EcoNet component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up EcoNet as config entry."""
+    hass.data[DOMAIN] = {}
```

Comment:
Use `dict.setdefault`. We allow multiple config entries so we can't overwrite hass.data for the domain.

Reply:
Can I just move this to the async_setup method?

Reply:
Yes, if we don't remove the dicts when unloading an entry.